internal/controller/groupqualityofservicelimits/groupqualityofservicelimits.go

```diff
@@ -198,11 +245,73 @@ func (c *external) Delete(ctx context.Context, mg resource.Managed) (managed.Ext
 		return managed.ExternalDelete{}, errors.New(errNotGroupQualityOfServiceLimits)
 	}
 
-	fmt.Printf("Deleting: %+v", cr)
+	cr.SetConditions(xpv1.Deleting())
+
+	user := cloudian.User{
+		GroupID: cr.Spec.ForProvider.GroupID,
+		UserID:  "*",
+	}
+	err := c.cloudianService.DeleteQOS(ctx, user, cr.Spec.ForProvider.Region)
+	// For consistency/futureproofing - API currently returns ok for non-existent groups
+	if err != nil && !errors.Is(err, cloudian.ErrNotFound) {
+		return managed.ExternalDelete{}, errors.Wrap(err, errGetCreds)
+	}
 
 	return managed.ExternalDelete{}, nil
 }
 
 func (c *external) Disconnect(ctx context.Context) error {
 	return nil
 }
+
+func toCloudianQos(qos v1alpha1.GroupQualityOfServiceLimitsParameters) (cloudian.QualityOfService, error) {
+	warning, err := toCloudianLimits(qos.Warning)
+	if err != nil {
+		return cloudian.QualityOfService{}, err
+	}
+	high, err := toCloudianLimits(qos.Hard)
+	if err != nil {
+		return cloudian.QualityOfService{}, err
+	}
+	return cloudian.QualityOfService{
+		Warning: warning,
+		Hard:    high,
+	}, nil
+}
+
+func toKiB(q *apir.Quantity) (int64, error) {
+	i, ok := q.AsInt64()
+	if !ok {
+		i, ok = q.AsDec().Unscaled()
+		if !ok {
+			return 0, errors.New("Unable to convert quantity to KiB")
+		}
+	}
+	if i > 0 {
+		return i / 1024, nil
```

Comment:
How does Go do trunc/round? I must admit I have never worked with floating point types in Go yet. 😆

Reply:
-1 / 1024 = 0
-1023 / 1024 = 0
-1024 / 1024 = -1
0 / 1024 = 0
1024 / 1024 = 1
2047 / 1024 = 1
2048 / 1024 = 2